scapy/layers/inet6.py

```diff
@@ -382,7 +382,7 @@ def hashret(self):
 
         if conf.checkIPsrc and conf.checkIPaddr and not in6_ismaddr(sd):
             sd = inet_pton(socket.AF_INET6, sd)
-            ss = inet_pton(socket.AF_INET6, self.src)
+            ss = inet_pton(socket.AF_INET6, ss)
```

Comment:
It's because of this change.
`ss` was unused and it made sense in the program.

However I haven't read the IPv6 RFC so I'm unsure of what it does

Reply:
I think that it is safer to keep the current value =)

Reply:
Should I delete the rest of the dead code then ?
https://github.com/secdev/scapy/blob/236102447f41f9818a09cde91c83af472b9b7383/scapy/layers/inet6.py#L342
https://github.com/secdev/scapy/blob/236102447f41f9818a09cde91c83af472b9b7383/scapy/layers/inet6.py#L381

Reply:
I had a look at the code and your initial fix is correct.